lib/membrane_webm/demuxer.ex

```diff
@@ -0,0 +1,296 @@
+defmodule Membrane.WebM.Demuxer do
+  @moduledoc """
+  Filter element for demuxing WebM files.
+
+  Receives a bytestream in WebM file format as input and outputs the constituent tracks of that file
+  onto seperate pads.
+
+  Works in three phases:
+
+  - :reading_header
+    Demands and parses the beginning bytes of the WebM file describing it's contents and sends:
+    `{:notify, {:new_track, {track_id, track_t}}}`
+    message to the parent pipeline for every track contained in the file.
+
+  - :notified_about_content
+    Pauses and waits for an output pad to be linked for every track in the file.
+    Expects elements to be linked via pad `Pad.ref(:output, track_id)`.
+
+  - :all_outputs_linked
+    Once all the expected output pads are linked it starts streaming the file's media with speed adjusted
+    to the slowest of the output elements i.e. pausing when demands on that element's pad equals 0.
+
+  WebM files can contain many:
+    - VP8 or VP9 video tracks
+    - Opus or Vorbis audio tracks
+
+  This module doesn't support Vorbis encoding.
+  """
+  use Membrane.Filter
+
+  alias Membrane.{Buffer, Time, Pipeline.Action}
+  alias Membrane.{Opus, RemoteStream, VP8, VP9}
+
+  def_input_pad :input,
+    availability: :always,
+    mode: :pull,
+    demand_unit: :buffers,
+    caps: {RemoteStream, content_format: :WEBM}
+
+  def_output_pad :output,
+    availability: :on_request,
+    mode: :pull,
+    caps: [VP8, VP9, Opus]
+
+  defmodule State do
+    @moduledoc false
+    @type track_t :: audio_track_t | video_track_t
+
+    @type audio_track_t :: %{
+            codec: :opus,
+            uid: non_neg_integer,
+            active: boolean,
+            channels: non_neg_integer
+          }
+
+    @type video_track_t :: %{
+            codec: :vp8 | :vp9,
+            uid: non_neg_integer,
+            active: boolean,
+            height: non_neg_integer,
+            width: non_neg_integer,
+            rate: Membrane.Time.t(),
+            scale: non_neg_integer
```

Comment:
These two are a little confusing. `rate` looks a little bit like timescale, but then, there is scale... I think we need to clarify that